tests/bgp/test_traffic_shift.py

```diff
@@ -0,0 +1,275 @@
+TS_ERROR = "System Mode: Not consistent"
+
+DUMP_FILE = "/tmp/bgp_monitor_dump.log"
+CUSTOM_DUMP_SCRIPT = "bgp/bgp_monitor_dump.py"
+CUSTOM_DUMP_SCRIPT_DEST = "/usr/share/exabgp/bgp_monitor_dump.py"
+BGP_MONITOR_PORT = 7000
+BGP_MONITOR_NAME = "bgp_monitor"
+BGP_ANNOUNCE_TIME = 30 #should be enough to receive and parse bgp updates
+
+# TODO: remove me
+BGPMON_TEMPLATE_FILE = 'bgp/templates/bgpmon.j2'
+BGPMON_CONFIG_FILE = '/tmp/bgpmon.json'
+
+@pytest.fixture
+def traffic_shift_community(duthost):
+    community = duthost.shell('sonic-cfggen -y /etc/sonic/constants.yml -v constants.bgp.traffic_shift_community')['stdout']
+    return community
+
+@pytest.fixture
+def common_setup_teardown(ptfhost, duthost, localhost):
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    dut_lo_addr = mg_facts['minigraph_lo_interfaces'][0]['addr']
+    dut_mgmt_ip = mg_facts['minigraph_mgmt_interface']['addr']
+    asn = mg_facts['minigraph_bgp_asn']
+    peer_addr = ptfhost.mgmt_ip
+    # TODO: Add a common method to load BGPMON config for test_bgpmon and test_traffic_shift
+    logger.info("Configuring bgp monitor session on DUT")
+    bgpmon_args = {
+        'peer_addr': peer_addr,
+        'asn': asn,
+        'local_addr': dut_lo_addr,
+        'peer_name': BGP_MONITOR_NAME
+    }
+    bgpmon_template = Template(open(BGPMON_TEMPLATE_FILE).read())
+    duthost.copy(content=bgpmon_template.render(**bgpmon_args),
+                 dest=BGPMON_CONFIG_FILE)
+
+    # Add a static route via mgmt interface on ptf
+    ptfhost.shell('ip route add {}/32 via {} dev mgmt'.format(dut_lo_addr, dut_mgmt_ip))
+    logger.info("Starting bgp monitor session on PTF")
+    ptfhost.file(path=DUMP_FILE, state="absent")
+    ptfhost.copy(src=CUSTOM_DUMP_SCRIPT, dest=CUSTOM_DUMP_SCRIPT_DEST)
+    ptfhost.exabgp(name=BGP_MONITOR_NAME,
+                   state="started",
+                   local_ip=peer_addr,
+                   router_id=peer_addr,
+                   peer_ip=dut_lo_addr,
+                   local_asn=asn,
+                   peer_asn=asn,
+                   port=BGP_MONITOR_PORT,
+                   dump_script=CUSTOM_DUMP_SCRIPT_DEST)
+    pytest_assert(wait_tcp_connection(localhost, ptfhost.mgmt_ip, BGP_MONITOR_PORT),
+                  "Failed to start bgp monitor session on PTF")
+    yield
+    # Cleanup bgp monitor
+    duthost.shell("redis-cli -n 4 -c DEL 'BGP_MONITORS|{}'".format(peer_addr))
+    ptfhost.exabgp(name=BGP_MONITOR_NAME, state="absent")
+    ptfhost.file(path=CUSTOM_DUMP_SCRIPT_DEST, state="absent")
+    ptfhost.file(path=DUMP_FILE, state="absent")
+    ptfhost.shell('ip route flush {}/32'.format(dut_lo_addr))
+
+def get_traffic_shift_state(host):
+    outputs = host.shell('TSC')['stdout_lines']
+    for out in outputs:
+        if TS_NORMAL == out.strip():
+            return TS_NORMAL
+        if TS_MAINTENANCE == out.strip():
+            return TS_MAINTENANCE
+    return TS_ERROR
```

Comment:
Here I'd check for TS_ERROR explicitly and raise an error if both of three outputs are not there

Reply:
Great suggestion. Updated.